Assets/MirageXR/DataModel/DataModel.cs

```diff
@@ -0,0 +1,319 @@
+﻿#nullable enable
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using UnityEngine;
+
+namespace DataModel
+{
+    [Serializable]
+    public class DataModel
+    {
+        [JsonProperty] public Guid Id { get; set; }
+        [JsonProperty] public DateTime CreationDate { get; set; }
+        [JsonProperty] public string Version { get; set; }
+    }
+
+    [Serializable]
+    public class Activity : DataModel
+    {
+        [JsonProperty] public User Creator { get; set; }
+        [JsonProperty] public string Name { get; set; }
+        [JsonProperty] public string Description { get; set; }
+        [JsonProperty] public string Language { get; set; } //is it needed?
+        [JsonProperty] public File Thumbnail { get; set; }
+        [JsonProperty] public StepHierarchy Hierarchy { get; set; }
+        [JsonProperty] public List<Content> Content { get; set; }
+        [JsonProperty] public List<ActivityStep> Steps { get; set; }
+    }
+
+    [Serializable]
+    public class StepHierarchy
+    {
+        [JsonProperty] public List<HierarchyItem> Item { get; set; }
+    }
+
+    [Serializable]
+    public class HierarchyItem
+    {
+        [JsonProperty] public List<Guid>? StepIds { get; set; }
+        [JsonProperty] public List<HierarchyItem>? Item { get; set; }
+        [JsonProperty] public string Title { get; set; }
+        [JsonProperty] public string Description { get; set; }
+    }
+
+    [Serializable]
+    public class ActivityStep : DataModel
+    {
+        [JsonProperty] public uint Order { get; set; } //or it can be based on Activity.Steps?
+        [JsonProperty] public string Name { get; set; }
+        [JsonProperty] public string Description { get; set; }
+        [JsonProperty] public List<Guid> Contents { get; set; } //Content.Id
+        [JsonProperty] public Location Location { get; set; }
+        [JsonProperty] public List<Trigger>? Triggers { get; set; } // cannot be null?
+        [JsonProperty] public List<Comment> Comments { get; set; }
+        [JsonProperty] public List<RequiredToolsPartsMaterial> RequiredToolsPartsMaterials { get; set; }
+        [JsonProperty] public List<File>? Attachment { get; set; }
+        [JsonProperty] public List<Comment> PrivateNotes { get; set; }
+    }
+
+    [Serializable]
+    public class Comment : DataModel
+    {
+        [JsonProperty] public User User { get; set; }
+        [JsonProperty] public string Text { get; set; }
+        [JsonProperty] public List<File>? Attachment { get; set; }
+    }
+
+
+    [Serializable]
+    public class RequiredToolsPartsMaterial
```

Comment:
should be `RequiredToolsPartsMaterials` (plural at the end)

Reply:
done